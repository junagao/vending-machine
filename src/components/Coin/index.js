export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
