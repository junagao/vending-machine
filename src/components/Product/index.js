export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
