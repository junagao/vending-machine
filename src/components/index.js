export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
