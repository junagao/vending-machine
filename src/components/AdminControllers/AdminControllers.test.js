import React from 'react';
import { shallow } from 'enzyme';

import { AdminControllers, RefillCoins, RefillStock } from 'components';

describe('AdminControllers', () => {
  let wrapper;

  const coinsMock = [
    { id: 'coin1', name: '2 euros', value: 2, quantity: 2 },
    { id: 'coin2', name: '1 euro', value: 1, quantity: 10 },
  ];
  const productsMock = [
    {
      id: 'coke',
      name: 'coke',
      img: 'image/coke.png',
      price: 1,
      currentStock: 10,
    },
    {
      id: 'sprite',
      name: 'sprite',
      img: 'image/sprite.png',
      price: 1.25,
      currentStock: 5,
    },
  ];

  beforeEach(() => {
    wrapper = shallow(
      <AdminControllers
        coins={coinsMock}
        onRefillCoinsQuantity={jest.fn()}
        products={productsMock}
        onRefillProductStock={jest.fn()}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one instance of <RefillCoins />', () => {
    expect(wrapper.find(RefillCoins).length).toEqual(1);
  });

  it('should render one instance of <RefillStock />', () => {
    expect(wrapper.find(RefillStock).length).toEqual(1);
  });
});
