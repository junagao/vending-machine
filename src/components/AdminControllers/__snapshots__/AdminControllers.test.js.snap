// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminControllers should render correctly given the required props 1`] = `
<Fragment>
  <RefillCoins
    coins={
      Array [
        Object {
          "id": "coin1",
          "name": "2 euros",
          "userQuantity": 2,
          "value": 2,
        },
        Object {
          "id": "coin2",
          "name": "1 euro",
          "userQuantity": 10,
          "value": 1,
        },
      ]
    }
    onRefillCoinsQuantity={[MockFunction]}
  />
  <RefillProductStock
    onRefillProductStock={[MockFunction]}
    products={
      Array [
        Object {
          "currentStock": 10,
          "id": "coke",
          "img": "image/coke.png",
          "name": "coke",
          "price": 1,
        },
        Object {
          "currentStock": 5,
          "id": "sprite",
          "img": "image/sprite.png",
          "name": "sprite",
          "price": 1.25,
        },
      ]
    }
  />
  <ul
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <li
      key="coin1"
    >
      <p>
        2 euros
      </p>
      <p />
    </li>
    <li
      key="coin2"
    >
      <p>
        1 euro
      </p>
      <p />
    </li>
  </ul>
</Fragment>
`;
